src/wp-includes/formatting.php

```diff
@@ -2017,8 +2017,9 @@ function sanitize_file_name( $filename ) {
 	 */
 	$special_chars = apply_filters( 'sanitize_file_name_chars', $special_chars, $filename_raw );
 
+	$filename = str_replace( '%20', '-', $filename );
```

Comment:
```suggestion
	$substitute_for_hyphen = array( '%20', '+', ' ' /* maybe others */, '-' );
	$special_chars = apply_filters( 'sanitize_file_name_chars', $special_chars, $filename_raw );
	$substitute_for_hyphen = apply_filters( 'sanitize_file_name_hyphenate', $substitute_for_hyphen, $filename_raw );

	$filename = str_replace( $substitute_for_hyphen, '-', $filename );
```

I checked in with some security minded folk to make sure we weren't missing something with changing this code.

As `+` is considered a special character, `str_replace( '+', '-', $filename );` would never perform any replacements as the plus sign has been removed by then.

I am wondering if there ought to be an array of special characters to be replaced by a hyphen. The above suggestion includes a couple but I think others could be included, for example `()` seem like suitable targets.